sale_elaboration/models/sale_order.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+def _execute_onchanges(records, field_name):
+    """Helper methods that executes all onchanges associated to a field."""
+    for onchange in records._onchange_methods.get(field_name, []):
+        for record in records:
+            record._origin = record.env['sale.order.line']
+            onchange(record)
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _create_elaboration_line(self, product, qty):
+        """
+        Create a sale order line from a elaboration product, search a line
+        with the same elaboration product to add qty
+        :param product:
+        :param qty:
+        :return: the sale order line record created
+        """
+        SaleOrderLine = self.env['sale.order.line']
+        sol_for_product = self.order_line.filtered(
+            lambda x: x.product_id == product)
+        if sol_for_product:
+            sol_for_product.product_uom_qty += qty
+            return sol_for_product
+        sol = SaleOrderLine.new({
+            'order_id': self.id,
+            'product_id': product.id,
+            'is_elaboration': True,
+        })
+        sol._origin = SaleOrderLine
```

Comment:
When i did the test and I call to onchange method the decorator is not executed, so I do not hav the attribute _origin.
Perhaps this would be solved by other way...

Reply:
The only reference I find for this is https://github.com/odoo/odoo/blob/f8b38f64d3cad59228b9ab4bbc32507d915c6b7d/addons/sale_stock/models/sale_order.py#L193, and in this case it doesn't add anything valuable.